File: backend/src/domains/analytics/repositories/AnalyticsRepository.ts
import type { DailyStats } from "../entities/DailyStats";
import type { StudyStreak } from "../entities/StudyStreak";
import type { WeaknessArea } from "../entities/WeaknessArea";

export interface AnalyticsRepository {
  getDailyStats(userId: string): Promise<DailyStats[]>;
  getWeaknessAreas(userId: string): Promise<WeaknessArea[]>;
  getStudyStreak(userId: string): Promise<StudyStreak | null>;
}

File: apps/mobile/lib/shared/repositories/analytics_repository.dart
import '../models/analytics.dart';
import '../services/api_client.dart';

class AnalyticsRepository {
  final ApiClient _api;

  const AnalyticsRepository(this._api);

  Future<List<DailyStats>> getDailyStats({String period = '30d'}) async {
    final Map<String, dynamic> queryParams = {
      'period': period,
    };
    final response = await _api.get('/api/analytics/stats', query: queryParams);
    final items = listFrom(response);
    return items.map((item) => DailyStats.fromJson(item)).toList();
  }

  Future<List<WeaknessArea>> getWeaknesses() async {
    final response = await _api.get('/api/analytics/weaknesses');
    final items = listFrom(response);
    return items.map((item) => WeaknessArea.fromJson(item)).toList();
  }

  Future<Map<String, dynamic>> getRecommendations() async {
    final response = await _api.get('/api/analytics/recommendations');
    return unwrap(response);
  }
}
